src/Compilers/CSharp/Portable/Parser/LanguageParser_Patterns.cs

```diff
@@ -595,20 +529,20 @@ SyntaxKind.GreaterThanToken or
                     SyntaxKind.GreaterThanEqualsToken;
         }
 
-        private PostSkipAction SkipBadPatternListTokens<T>(
-            ref SyntaxToken open,
-            SeparatedSyntaxListBuilder<T> list,
-            SyntaxKind expected,
-            SyntaxKind closeKind)
+        private static PostSkipAction SkipBadPatternListTokens<T>(
+            LanguageParser @this, ref SyntaxToken open, SeparatedSyntaxListBuilder<T> list, SyntaxKind expectedKind, SyntaxKind closeKind)
             where T : CSharpSyntaxNode
         {
-            return this.SkipBadSeparatedListTokensWithExpectedKind(ref open, list,
-                p => p.CurrentToken.Kind != SyntaxKind.CommaToken && !p.IsPossibleSubpatternElement(),
-                p => p.CurrentToken.Kind == closeKind || p.CurrentToken.Kind == SyntaxKind.SemicolonToken || p.IsTerminator(),
-                expected);
+            if (@this.CurrentToken.Kind is SyntaxKind.CloseParenToken or SyntaxKind.CloseBraceToken or SyntaxKind.CloseBracketToken or SyntaxKind.SemicolonToken)
+                return PostSkipAction.Abort;
+
+            return @this.SkipBadSeparatedListTokensWithExpectedKind(ref open, list,
+                static p => p.CurrentToken.Kind != SyntaxKind.CommaToken && !p.IsPossibleSubpatternElement(),
+                static (p, closeKind) => p.CurrentToken.Kind == closeKind || p.CurrentToken.Kind == SyntaxKind.SemicolonToken,
+                expectedKind, closeKind);
         }
 
-        private ExpressionSyntax ParseSwitchExpression(ExpressionSyntax governingExpression, SyntaxToken switchKeyword)
+        private SwitchExpressionSyntax ParseSwitchExpression(ExpressionSyntax governingExpression, SyntaxToken switchKeyword)
```

Comment:
I assume you made a deliberate decision to not revise parsing switch expression arms here. Do we have a note somewhere to follow up on it? Assuming it would be a benefit to unify switch expression arm parsing with the other comma-separated lists.

Reply:
I didn't update code that wasn't using the existing pattern.  We can certainly go back and attempt to do that if someone wants :)  What i was trying to do was have a common helper for all the cases that had been copy/pasted and slowly changed over time :)

Reply:
as an example, this code doesn't use the pattern of IsSwitchExpressionArm/ParseSwitchExpressionArm.  So it's a non-trivial update to move from this form to teh form the rest have.  Why this code doesn't follow the other pattern is something i don't have understanding of.

Reply:
put another way ParseCommaSeparatedList is the helper for all separated-list-parsing that has the same breakdown of "IsXXX, ParseXXX, SkipBadXXX".  If we have such code, it should move to the helper.